[Test] Creating CADMIN 1.15 python test module

j-ororke opened the pull request:
- Creating TC_CADMIN_1_15.py test module following test steps in draft PR from Cecille here: [4636](https://github.com/CHIP-Specifications/chip-test-plans/pull/4636)
- Removing test_TC_CADMIN_1_15.yaml as no longer needed once python module is created.

#### Testing

j-ororke committed:
Created CADMIN 1.15 test module:
- Creating TC_CADMIN_1_15.py test module following test steps in draft PR from Cecille here: https://github.com/CHIP-Specifications/chip-test-plans/pull/4636
- Removing test_TC_CADMIN_1_15.yaml as no longer needed once python module is created.

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_CADMIN_1_15 test module:
- Fixing linting errors

j-ororke committed:
Updated TC_CADMIN_1_15 test module:
- Made minor linting fixes

j-ororke committed:
Updated TC_CADMIN_1_15 test module:
- Updated comment verbage to correct information
- Updated and removed some expected results for test steps as necessary

restyled-commits committed:
Restyled by autopep8

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+
+
+class TC_CADMIN_1_15(MatterBaseTest):
+    async def OpenCommissioningWindow(self, th, expectedErrCode) -> CommissioningParameters:
```
see comment on CADMIN-1.11 - it seems clearer here to have a None error code. Also add the type annotations.

j-ororke replied:
This should now be resolved, thank you

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+
+
+class TC_CADMIN_1_15(MatterBaseTest):
+    async def OpenCommissioningWindow(self, th, expectedErrCode) -> CommissioningParameters:
```
Given that this function appears in a couple of CADMIN tests, it seems like it would make sense to consolidate. Let's get these all in as-is, and then pull them into a single file with a single implementation of this function. The bonus there is we can run them all in one go and save some time at testing. Well, not through the TH, but in the CI.

j-ororke replied:
Created a follow-up task here: [514](https://github.com/project-chip/matter-test-scripts/issues/514)

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+            nodeId=self.dut_node_id, setupPinCode=setupPinCode,
+            filterType=ChipDeviceCtrl.DiscoveryFilterType.LONG_DISCRIMINATOR, filter=self.discriminator)
+
+    def generate_unique_random_value(self, value):
```
does this function get used anywhere? left over from something else?

j-ororke replied:
Sorry, I will remove this, it was left over in the template from when TC_CADMIN_1_9 test module was created.

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+
+    def steps_TC_CADMIN_1_15(self) -> list[TestStep]:
+        return [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
```
something's wacky with the step numbering here - doesn't quite match the test plan, i don't think. Would you mind double checking?

cecille replied:
https://github.com/CHIP-Specifications/chip-test-plans/pull/4636/files

j-ororke replied:
the test steps in the test script should now match the test plan

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+            asserts.fail("Expected number of fabrics not correct")
+
+        if 2 in fabric_indexes2:
+            asserts.fail("fabricIndexes should consist of indexes 1, 3, and 4 at this time")
```
Can you change the assertion comment here? The index numbers aren't guaranteed to start from 1

j-ororke replied:
Thank you for catching this, I have updated the asertion message to report that the fabric indexes should not contain the fabric_idx_2 set value

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+            # len of fabrics is expected to be 2, if 2 not found then we assert failure
+            asserts.fail(f"Expected number of fabrics not correct, should show 2, but instead shows {str(len(fabrics2))}")
+
+        if 2 in fabric_indexes:
```
```suggestion
        if fabric_idx_cr2 in fabric_indexes:
```

cecille commented on src/python_testing/TC_CADMIN_1_15.py:
```diff
@@ -0,0 +1,244 @@
+        fabric_indexes = [fabric.fabricIndex for fabric in fabrics2]
+        if len(fabrics2) != initial_number_of_fabrics + 1:
+            # len of fabrics is expected to be 2, if 2 not found then we assert failure
+            asserts.fail(f"Expected number of fabrics not correct, should show 2, but instead shows {str(len(fabrics2))}")
```
I don't actually care too much, but you don't need str operator in an f string. less typing.

j-ororke replied:
Thank you, I will remember and change this!

cecille reviewed (commented):
minor comments, looks good, just need to wait on the dns-sd stuff.

j-ororke committed:
Apply suggestions from code review

Added changes from Ceciile to this test module

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Update TC_CADMIN_1_15 test module

- Added type annotations for OpenCommissioningWindow function
- Added None type expectedErrorCode variable
- Updated test step numbering to match test plan

j-ororke committed:
Update TC_CADMIN_1_15 test module:

- Updated test step 15 failure assertion to not start from number 1, but instead to correctly make sure that fabric_idx_cr2 does not exist in current fabric indexes list

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_15_test

j-ororke committed:
Updating TC_CADMIN_1_15 test module:
- Implementing DNS-SD feature from Raul to cover test step 8

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_CADMIN_1_15 test module:
- Resolving linting errors

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_15_test

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_15_test

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_CADMIN_1_15 test module:
- Changing method for test step 8 as needed

j-ororke committed:
Updating TC_CADMIN_1_15 and mdns_discovery python modules:
- Changed discovery to return last entry instead of first entry when run multiple times
- Updated TC_CADMIN_1_15 to include Raul's latest changes to mdns_discovery in order to complete test step 8

j-ororke committed:
Updating TC_CADMIN_1_15 test module:
- Resolving linting error

j-ororke committed:
Updating mdns_discovery module:
- To change to searching discovered_services dictionary values for expected mdns value instead of taking last or first value in list.

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_15_test

raul-marquez-csa commented on src/python_testing/mdns_discovery/mdns_discovery.py:
```diff
@@ -493,7 +494,8 @@ def _get_instance_name(self, service_info: AsyncServiceInfo) -> str:
                            log_output: bool,
-                           discovery_timeout_sec: float
+                           discovery_timeout_sec: float,
+                           expected_value: str = None,
```
Please add this new argument to the description

j-ororke replied:
Thank you, this has been updated in the latest code changes here now

raul-marquez-csa commented:
Please remove the Test_TC_CADMIN_1_15 line from manualTests.json

j-ororke committed:
Updating manualTests.json and mdns_discovery.py:
- Updated manualTests.json to remove Test_TC_CADMIN_1_15 from the list of CADMIN yaml scripts
- Updated mdns_discovery module to include description of "expected_value" arg.

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by prettier-json

j-ororke committed:
Merge branch 'master' into create_CADMIN_1_15_test
